src/controller/tests/data_model/TestRead.cpp

```diff
@@ -141,19 +147,22 @@ CHIP_ERROR ReadSingleClusterData(const Access::SubjectDescriptor & aSubjectDescr
         if (aPath.mClusterId == app::Clusters::UnitTesting::Id &&
             aPath.mAttributeId == app::Clusters::UnitTesting::Attributes::Int16u::Id)
         {
-            AttributeValueEncoder::AttributeEncodeState state =
-                (apEncoderState == nullptr ? AttributeValueEncoder::AttributeEncodeState() : *apEncoderState);
-            AttributeValueEncoder valueEncoder(aAttributeReports, aSubjectDescriptor.fabricIndex, aPath,
-                                               kDataVersion /* data version */, aIsFabricFiltered, state);
+            AttributeEncodeState state;
+            if (apEncoderState != nullptr)
+            {
+                state = *apEncoderState;
```

Comment:
OK, we have a _lot_ of copies of this pattern.  Is it worth having a `static AttributeEncodeState CopyState(AttributeEncodeState *)` or something which will do this in one place?  Don't feel strongly about this one.

Reply:
I created a constructor from a pointer, since this seems to be the common use case.